app/controllers/snippets-management/edit-snippet.js

```diff
@@ -0,0 +1,241 @@
+      code_block,
+
+      text,
+
+      image,
+
+      hard_break,
+      block_rdfa,
+      table_of_contents: table_of_contents(this.config.tableOfContents),
+      invisible_rdfa,
+      link: link(this.config.link),
+    },
+    marks: {
+      inline_rdfa,
+      em,
+      strong,
+      underline,
+      strikethrough,
+      subscript,
+      superscript,
+      highlight,
+      color,
+    },
+  });
+
+  get insertVariableWidgetOptions() {
+    const config = getOwner(this).resolveRegistration('config:environment');
+    return {
+      publisher: this.currentSession.group.uri,
+      defaultEndpoint: config.insertVariablePlugin.endpoint,
+      variableTypes: ['text', 'number', 'date', 'codelist'],
+    };
+  }
+
+  get config() {
+    const ENV = getOwner(this).resolveRegistration('config:environment');
+    return {
+      tableOfContents: [
+        {
+          nodeHierarchy: [
+            'title|chapter|section|subsection|article',
+            'structure_header|article_header',
+          ],
+        },
+      ],
+      date: {
+        placeholder: {
+          insertDate: this.intl.t('date-plugin.insert.date'),
+          insertDateTime: this.intl.t('date-plugin.insert.datetime'),
+        },
+        formats: [
+          {
+            label: 'Short Date',
+            key: 'short',
+            dateFormat: 'dd/MM/yy',
+            dateTimeFormat: 'dd/MM/yy HH:mm',
+          },
+          {
+            label: 'Long Date',
+            key: 'long',
+            dateFormat: 'EEEE dd MMMM yyyy',
+            dateTimeFormat: 'PPPPp',
+          },
+        ],
+        allowCustomFormat: true,
+      },
+      variable: {
+        publisher: this.currentSession.group?.uri,
+        defaultEndpoint: ENV.insertVariablePlugin.endpoint,
+        variableTypes: ['text', 'number', 'date', 'codelist'],
+      },
+      structures: STRUCTURE_SPECS,
+      citation: {
+        type: 'nodes',
+        activeInNodeTypes(schema) {
+          return new Set([schema.nodes.doc]);
+        },
+        endpoint: '/codex/sparql',
+      },
+      link: {
+        interactive: true,
+      },
+    };
+  }
+
+  get nodeViews() {
+    return (controller) => {
+      return {
+        variable: variableView(controller),
+        table_of_contents: tableOfContentsView(this.config.tableOfContents)(
+          controller
+        ),
+        link: linkView(this.config.link)(controller),
+      };
+    };
+  }
+
+  get plugins() {
+    return [
+      tablePlugin,
+      this.citationPlugin,
+      linkPasteHandler(this.schema.nodes.link),
+    ];
+  }
+
+  @action
+  handleRdfaEditorInit(editor) {
+    this.editor = editor;
+    if (this.editorDocument.content) {
+      editor.setHtmlContent(this.editorDocument.content);
+    }
+  }
+
+  get dirty() {
+    return this.editorDocument.content !== this.editor.htmlContent;
+  }
+
+  fetchEditorDocument = task(async () => {
+    this._editorDocument = await this.model.currentVersion;
+  });
```

Comment:
I could show some loading but in practice it was that fast, that I couldn't notice it, so I thought there wasn't needed.
I did it here instead of in the route, because it also has to work with the links, and the model is not called with those as it's provided